src/app/core/products/product-api.service.ts

```diff
@@ -51,13 +47,20 @@ export class ProductApiService extends DataApiService<Product> {
     }
 
     getProductsByType(productType?: ProductType | "all"): Observable<Product[]> {
-        if (!productType || productType === "all") {
-            return this.getProducts();
+        switch (productType) {
+            case undefined: /* fallthrough */
```

Comment:
is this needed?

fallthrough will just fall to the default case no? Also, the default case in this example should be all

In any case, use a hashmap. Switch statements have O(n) complexity, while objects (with unique keys) are O(1)

I prefer) use an object (e.g dict) where the keys are the product types (all, coffee, etc.) the values are the returned observables.

And then simply return dict[productType] || dict["all"].